src/shared/sam/localLambdaRunner.ts

```diff
@@ -462,10 +641,17 @@ export async function makeConfig(config: SamLaunchRequestArgs): Promise<void> {
     await writeFile(config.envFile, env)
 
     // event.json
-    if (config.lambda?.payload?.path) {
-        const fullpath = tryGetAbsolutePath(config.workspaceFolder, config.lambda?.payload?.path)
+    const payloadObj = config.lambda?.payload?.json ?? config.api?.payload?.json
+    const payloadPath = config.lambda?.payload?.path ?? config.api?.payload?.path
```

Comment:
Is there a reason why `lambda` is prioritized over `api`?

Reply:
No. Unsure what to do with this.

Reply:
Unsure about the full code path, but will `config` know the `target` type? If so, use that here. The worst case scenario is someone has their `target=api` launch config misconfigured with both values and they're confused as to why their payload isn't being used.

Reply:
Offline: we can address this later since this PR is pretty unwieldy.